File: problem-3/main.java
class ProblemThree {
    public static void main(String[] args) {
        System.out.println(largest_prime_factor(600851475143l));
    }
    public static long largest_prime_factor(long n){
        int i = 2;
        while (i*i <= n){
            if (n % i != 0) {
                i++;
            }
            else {
                n /= i;
            }
        }
        return n;
    }
}

File: problem-3/main.cs
using System;

namespace ProjectEuler {
    class ProblemTwo {
        static void Main(string[] args) {      
            Console.WriteLine(LargestPrimeFactor(600851475143));
        }
        static long LargestPrimeFactor(long n) {
            int i = 2;
            while (i*i <= n){
                if (n % i != 0){
                    i++;
                }
                else {
                    n /= i;
                }
            }
            return n;
        }
    }
}